x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rules.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import { useState, useCallback } from 'react';
+import { isEmpty } from 'lodash';
+import { Rule, Pagination } from '../../types';
+import { loadRules, LoadRulesProps } from '../lib/rule_api';
+import { useKibana } from '../../common/lib/kibana';
+
+interface RuleState {
+  isLoading: boolean;
+  data: Rule[];
+  totalItemCount: number;
+}
+
+type UseLoadRulesProps = Omit<LoadRulesProps, 'http'> & {
+  hasAnyAuthorizedRuleType: boolean;
+  onPage: (pagination: Pagination) => void;
+  onError: (message: string) => void;
+};
+
+export function useLoadRules({
+  page,
+  searchText,
+  typesFilter,
+  actionTypesFilter,
+  ruleExecutionStatusesFilter,
+  ruleStatusesFilter,
+  tagsFilter,
+  sort,
+  hasAnyAuthorizedRuleType,
+  onPage,
+  onError,
+}: UseLoadRulesProps) {
+  const { http } = useKibana().services;
+
+  const [rulesState, setRulesState] = useState<RuleState>({
+    isLoading: false,
+    data: [],
+    totalItemCount: 0,
+  });
+
+  const [noData, setNoData] = useState<boolean>(true);
+  const [initialLoad, setInitialLoad] = useState<boolean>(true);
+
+  const internalLoadRules = useCallback(async () => {
+    if (!hasAnyAuthorizedRuleType) {
+      return;
+    }
+    setRulesState((prevRuleState) => ({ ...prevRuleState, isLoading: true }));
+    try {
+      const rulesResponse = await loadRules({
+        http,
+        page,
+        searchText,
+        typesFilter,
+        actionTypesFilter,
+        ruleExecutionStatusesFilter,
+        ruleStatusesFilter,
+        tagsFilter,
+        sort,
+      });
+      setRulesState({
+        isLoading: false,
+        data: rulesResponse.data,
+        totalItemCount: rulesResponse.total,
+      });
+
+      if (!rulesResponse.data?.length && page.index > 0) {
```

Comment:
can you explain me the reasoning? I think I am getting confuse about it

Reply:
I believe this logic was there before I moved it to a hook, I didn't want to break anything so I moved it over